Bottom: b451f32fb1fdc37a514289f58a1e8aa08c444bfe
Top:    efb401032cfed5a8daf3820cc35db51e8b7653e9
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 17:45:00 +0000

Refresh of fix-return-codes-from

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 9282eda..37f2218 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2991,8 +2991,9 @@ BOOLp
 poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
 {
   *dst = NULL;
-  if (src != NULL)
-    *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
+  if (src == NULL)
+    return TRUE; /* Copying a NULL POLYAREA is not a failure, return true */
+  *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
   if (*dst == NULL)
     return FALSE;
   (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
